core/include/drivers/imx/dcp.h

```diff
@@ -0,0 +1,186 @@
+/* SPDX-License-Identifier: BSD-2-Clause */
+/*
+ * Copyright 2020 NXP
+ */
+#ifndef __IMX_DCP_H__
+#define __IMX_DCP_H__
+
+#include <compiler.h>
+#include <tee_api_types.h>
+#include <types_ext.h>
+#include <util.h>
+
+#define DCP_SHA_BLOCK_SIZE    64
+#define DCP_AES128_BLOCK_SIZE 16
+#define DCP_AES128_KEY_SIZE   DCP_AES128_BLOCK_SIZE
+#define DCP_AES128_IV_SIZE    DCP_AES128_BLOCK_SIZE
+
+enum dcp_key_mode {
+	DCP_SRAM0 = 0,
+	DCP_SRAM1,
+	DCP_SRAM2,
+	DCP_SRAM3,
+	DCP_PAYLOAD,
+	DCP_OTP,
+};
+
+enum dcp_aes_mode {
+	DCP_ECB = 0,
+	DCP_CBC,
+};
+
+enum dcp_aes_op {
+	DCP_ENCRYPT = 0,
+	DCP_DECRYPT,
+};
+
+enum dcp_hash_config {
+	DCP_SHA1 = 0,
+	DCP_SHA256,
+};
+
+enum dcp_channel {
+	DCP_CHANN0 = 0,
+	DCP_CHANN1,
+	DCP_CHANN2,
+	DCP_CHANN3,
+	DCP_NB_CHANNELS,
+};
+
+struct dcp_descriptor {
```

Comment:
It seems that the layout of struct needs be in sync with the hardware. If so please add a note on that.

Reply:
Ok, will do